sentry-core/src/main/java/io/sentry/core/config/provider/JndiSupport.java

```diff
@@ -0,0 +1,26 @@
+package io.sentry.core.config.provider;
+
+
+/**
+ * A helper class to check whether JNDI is available in the current JVM.
+ */
+final class JndiSupport {
+//    private static final Logger logger = LoggerFactory.getLogger(JndiSupport.class);
+
+    private JndiSupport() {
+    }
+
+    /**
+     * Checks whether JNDI is available.
+     * @return true if JNDI is available, false otherwise
+     */
+    public static boolean isAvailable() {
+        try {
+            Class.forName("javax.naming.InitialContext", false, JndiSupport.class.getClassLoader());
```

Comment:
This isnt' very Android friendly.
We'd need a clear way to init without going through all these Java ways

Reply:
I'd say that none of the providers should be "executed" on Android, but just its own AndroidProvider.

it doesnt make sense to try to read system envs, locate the file etc if the only thing that makes sense for android is its own provider.
ideally, the options have a list of providers, and its empty by default, so depending on the integration, they add their own providers that make sense, something like that.
eg sentry-spring would have a system env, file system provider etc...
sentry-android would have only its own provider, that reads from the manifest

we'll also need a new module/package like io.sentry.config that has non android related things, I dont customers needing to add extra rules to not fail their build because of unknown classes, this happened with the older version of the SDK and lot of people complained.